api/src/main/java/org/pmiops/workbench/captcha/CaptchaVerificationServiceImpl.java

```diff
@@ -60,11 +80,17 @@ public boolean verifyCaptcha(String responseToken) throws ApiException {
     }
     String captchaHostname = response.getHostname();
     String uiUrl = configProvider.get().admin.loginUrl;
+    boolean usingTestCaptcha = configProvider.get().captcha.useTestCaptcha;
+    boolean captchaHostNameMatchUI = false;
 
-    // check if the UI URL has the host as send by Captcha Response
-    boolean captchaHostNameMatchUI =
-        String.format(urlPattern, captchaHostname).equals(uiUrl)
-            || String.format(localHostUrlPattern, captchaHostname).equals(uiUrl);
+    // Production/Stable should  not use google test Key and the domainName should match with the
+    // one return from Captcha Response
+    if (!usingTestCaptcha) {
+      // check if the UI URL has the host as send by Captcha Response
+      captchaHostNameMatchUI = String.format(urlPattern, captchaHostname).equals(uiUrl);
+    } else {
+      captchaHostNameMatchUI = captchaHostname.equals(googleTestHost);
```

Comment:
Should you just compare this to `config.server.uiBaseUrl`, or at least - extract the host from there ?

Reply:
i was initially checking the config.loginUrl to be one of test/login/stable but since it was basically checking just the config variable to constants i was not sure if it was adding anything.

Reply:
https://github.com/all-of-us/workbench/blob/fd85f5755b299cfa8ac1f092dc9752ca87ba4d44/api/src/main/java/org/pmiops/workbench/captcha/CaptchaVerificationServiceImpl.java L100